README.md

```diff
@@ -1,9 +1,536 @@
+PeripheralManagerService service = new PeripheralManagerService();
+Pwm pwm = service.openPwm("PWM1");
+pwm.setPwmDutyCycle(50.0); // square wave
+
+// Play a note
+int frequency = 440;
+pwm.setPwmFrequencyHz(frequency);
+pwm.setEnabled(true);
+
+// Wait
+Thread.sleep(500);
+
+// Stop playing the note
+pwm.setEnabled(false);
+
+// Close the device when done (onDestroy)
+pwm.close();
+```
+
+Now, integrate that code to your project, to play some sound when a button is pressed.
+
+**Question:** What does the following code do? Try it!
+```java
+int[] frequencies = new int[] { 261, 294, 329, 349, 392, 440, 493, 523 };
+pwm.setEnabled(true);
+for (int frequency : frequencies) {
+  pwm.setPwmFrequencyHz(frequencies[i]);
+  Thread.sleep(300);
+}
+pwm.setEnabled(false);
+```
+
+#### Making things simpler
+
+As for the button, there's an official [PWM speaker](https://github.com/androidthings/contrib-drivers/tree/master/pwmspeaker) driver that makes it easier to play tones
+
+```java
+Speaker buzzer = new Speaker("PWM1");
+buzzer.play(frequency);
+buzzer.stop();
+buzzer.close(); // onDestroy()
+```
+
+If you take a look at the PWM speaker driver [source code](https://github.com/androidthings/contrib-drivers/blob/master/pwmspeaker/src/main/java/com/google/android/things/contrib/driver/pwmspeaker/Speaker.java), you will be surprised by how small the implementation is.  
+It is actually very similar to what we did manually.
+
+If one day you have to create your own driver, you can see that it may be easier than you might think.
+
+
+## Using the servo motor
+
+Now that you are familiar with PWM, we will use the servo motor which also is a PWM component.  
+
+There is an official driver that simplifies using PWM servo motors.  
+We will use it; we already know how PWM works, and it will prevent us from writing too much code.
+
+### Setting up the Servo motor
+
+When creating a driver, or setting up a component. We have to read the documentation.  
+Take a look at the following picture, extracted from an official [SG90 documentation](http://www.micropik.com/PDF/SG90Servo.pdf)
+
+![servodoc][servodoc]
+
+We can see the following information:
+
+* A ~1ms pulse means moving the servo to -90
+* A ~2ms pulse means moving the servo to 90
+
+This translates into code like that:
+
+```java
+Servo servo = new Servo("PWM0");
+servo.setPulseDurationRange(1, 2);
+servo.setAngleRange(-90, 90);
+servo.setEnabled(true);
+```
+
+(again, don't forget to close the servo in the `onDestroy`)
+
+To make the servo move, we have to provide it an angle. Examples:
+
+```java
+servo.setAngle(42d);
+// or
+servo.setAngle(servo.getMinimumAngle());
+// or
+servo.setAngle(servo.getMaximumAngle());
+```
+
+Using this code snippet, move the servo to its minimum angle when pressing a button, and move it to its maximum angle when pressing the button again.  
+You'll probably have to create a boolean variable that keeps the state of the PWM (whether it is at its minimum or maximum angle).
+
```

Comment:
Maybe we could add a note about the fact that on all the SG90 servos, the size of the angle won't be 180° if you use the documentation's values (1ms-2ms). You will have only 90°.
If you want to have a wider angle (almost 180°) you need to use the range ~0.6ms - ~2.4ms. That's what I observed on all my SG90.

Here is my proposition to add:
_Note:_ The documentation values are not really good here. 1ms-2ms pulse duration will give a 90° angle instead of 180°. If you want to have a wider angle, use a wider range like 0.6ms to 2.4ms for example.